bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java

```diff
@@ -294,6 +296,22 @@ public Set<ConfigPropertyMetadata> configure()
         return configurationFactory.getUsedProperties();
     }
 
+    private String didYouMean(String key, Set<String> allSeenProperties)
+    {
+        ImmutableList.Builder<String> matches = ImmutableList.builder();
+        for (String seenProperty : allSeenProperties) {
+            if (FuzzySearch.ratio(seenProperty, key) > 80) {
+                matches.add(seenProperty);
+            }
+        }
+        List<String> matched = matches.build();
+        if (matched.isEmpty()) {
+            return "";
+        }
+
+        return ", did you mean to use " + matched.stream().map(value -> "'" + value + "'").collect(joining(" or ")) + "?";
```

Comment:
Please extract `matched.stream().map(value -> "'" + value + "'").collect(joining(" or "))` as variable.

How many things there is an collection? If there is more than two then multiple `or` may sound poorly. See `Did you mean to use a or b or c or d` vs `Did you mean a,b,c or d`?

Reply:
I've made it format up to 3 suggestions

Reply:
👍